app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/ImageDialogFragment.java

```diff
@@ -0,0 +1,98 @@
+package com.teamagam.gimelgimel.app.view.fragments.dialogs;
+
+import android.app.Activity;
+import android.app.Fragment;
+import android.net.Uri;
+import android.view.View;
+import android.widget.TextView;
+
+import com.facebook.drawee.view.SimpleDraweeView;
+import com.teamagam.gimelgimel.R;
+import com.teamagam.gimelgimel.app.model.ViewsModels.MessageImage;
+import com.teamagam.gimelgimel.app.model.entities.ImageMetadata;
+import com.teamagam.gimelgimel.app.view.drawable.CircleProgressBarDrawable;
+import com.teamagam.gimelgimel.app.view.fragments.dialogs.base.BaseDialogFragment;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+
+/**
+ * Represents a dialog of incoming image messages.
+ */
+public class ImageDialogFragment extends BaseDialogFragment {
+
+    private SimpleDraweeView mDraweeView;
+    private TextView mDateTextView;
+    private TextView mLatTextView;
+    private TextView mLonTextView;
+
+    private MessageImage mMessage;
+
+    /**
+     * Sets the message to show in the dialog.
+     * The method will parse the message and update the dialog values
+     * @param message - The current message to show
+     */
+    public void setImageMessage(MessageImage message) {
+        mMessage = message;
```

Comment:
this method must be called *after*  dialog.show().
it updates views.

fix it or add some checks. add comment. !fixed